executor/script.go

```diff
@@ -169,23 +179,54 @@ func RunScript(script Script, timeout int) ExecutionResult {
 					Help:   script.Help,
 				},
 			},
+			TotalExecTime: execTotalMs,
+		}
+	}
+
+	log.Debugf("Running %s with timeout of %v", script.Path, (time.Duration(int(timeout)) * time.Second))
+
+	// The following block of code is used instead of CommandContext as the context
+	// doesn't terminate sub-processes.
+	// See: https://github.com/golang/go/issues/22485
+
+	cmd := exec.Command("/bin/bash", "-c", script.Path)
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+	killChanRes := make(chan ExecutionResult, 1)
+
+	time.AfterFunc(time.Duration(int(timeout))*time.Second, func() {
+		syscall.Kill(-cmd.Process.Pid, syscall.SIGKILL)
+		duration := time.Since(execStart)
+		execTotalMs := duration.Nanoseconds() / 1000000
+		killChanRes <- ExecutionResult{
+			ScriptPath:    script.Path,
+			Error:         errors.New("Script execution timed out"),
+			TotalExecTime: execTotalMs,
+		}
+	})
+	output, outErr := cmd.Output()
+
+	for {
```

Comment:
I must be missing something because I find that cmd.Output() is blocking (even if running commands in the background within the script) so you only get there if one of these 3 conditions are met:
- the command runs and finishes normally
- the command returns an error (ex wrong args or whatever)
- your kill callback has been executed and manually terminated the command

So by the time we get here, we can know for sure if the command executed well, exited because of a script failure or has been terminated by the time-out-callback.

As a consequence, a single select statement would do the trick as we only need to check if the process was terminated by the callback:
```suggestion
	select {
	// Time out error
	case res := <- killChanRes:
	    return res
	
	// Go on if the process did not time out
	default:
	    break
        }
```